src/Sylius/Bundle/PromotionBundle/Behat/PromotionContext.php

```diff
@@ -128,6 +128,19 @@ public function theFollowingPromotionsExist(TableNode $table)
     }
 
     /**
+     * @Given :promotionName is coupon based promotion
+     */
+    public function isCouponBasedPromotion($promotionName)
+    {
+        $promotion = $this->findOneByName('promotion', $promotionName);
+        $promotion->setCouponBased(true);
+
+        $manager = $this->getEntityManager();
+        $manager->persist($promotion);
```

Comment:
If we use `find` on repository we don't have to persist, beacuse the entity is already in the unit of work.